### DISCUSSION TITLE
Both @mouseover & @click handler called when clicking on device without hover capability.

### DISCUSSION THREAD

**Discussion by mjauvin** [alpinejs/alpine#3827] [7. Bugs (archived)] [answered]
I have a link like below:
```
<a x-data={state=false} href="..." @click="state=!state" @mouseover="state=!state">My link</a>
```

It works fine when used on a device with mouse hover capability, but on mobile, when I click the link both handlers get called and cancel each other. I can work around the problem with special code, but I would expect the @mouseover to only trigger when actually hovering, not clicking.

**Comment by SimoTod** [comment, marked as answer]
Those are browser events, Alpine just listen for them. When you click on something on a mobile browser, it also triggers the mouseover event.

**Comment by ekwoka** [comment]
You would need to manually check the mousover event to see the pointer type to cut out touch events.

Or have both events point to a debounced toggle.

Really, this example would be bad design (i'm assuming your actual code is different?). if someone moved their mouse and clicked quickly you'd also have 2 events. Your code should handle these cases in an idempotent way.
